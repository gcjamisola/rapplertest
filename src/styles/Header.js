import styled from 'styled-components';

const NavContainer = styled.ul`
    text-align: right;
`;

const NavItem = styled.li`
    cursor: pointer;
    display: inline-block;
    list-style-type: none;
    margin-left: 10px;
    padding: 5px 10px;
`;

export { NavContainer, NavItem };