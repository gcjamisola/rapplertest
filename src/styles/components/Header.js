import styled from 'styled-components';

const NavContainer = styled.ul`
	background: #232d3b;
	display: flex;
	height: 50px;
	justify-content: flex-end;
	margin: 0;
	padding: 0;
`;

const NavItem = styled.li`
		color: aliceblue;
		cursor: pointer;
		border-bottom: none;
    justify-content: flex-end;
    list-style-type: none;
		margin: auto 0;
		margin-right: 1em;
		padding: 5px 10px;
		transition: border-bottom 0.2s ease;

		&:hover {
			border-bottom: 2px solid #61ccff;
		}
`;

export { NavContainer, NavItem };
