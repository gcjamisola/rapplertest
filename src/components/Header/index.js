import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { NavContainer, NavItem } from '../../styles';

class Header extends Component {
    render() {
        return (
            <div>
                <NavContainer>
                    <NavItem>Home</NavItem>
                    <NavItem>About</NavItem>
                    <NavItem>Gallery</NavItem>
                    <NavItem>Contact Us</NavItem>
                </NavContainer>
            </div>
        );
    }
}

export default Header;