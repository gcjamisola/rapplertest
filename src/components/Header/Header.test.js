import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Header from './index';

describe('<Header/>', () => {
    it('renders <Header/> component', () => {
        const component = shallow(<Header/>);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    })
})