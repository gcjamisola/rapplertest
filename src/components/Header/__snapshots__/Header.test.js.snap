// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders <Header/> component 1`] = `
<div>
  <styled.ul>
    <styled.li>
      Home
    </styled.li>
    <styled.li>
      About
    </styled.li>
    <styled.li>
      Gallery
    </styled.li>
    <styled.li>
      Contact Us
    </styled.li>
  </styled.ul>
</div>
`;
